File: Models/Monitoring.cs
using System;
using MtpApp.Models;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations.Schema;

namespace MtpApp.Models
{
    public class Monitoring
    {
        public int Id { get; set; }
        [Required]
        [StringLength(50)]
        public string MonitoringTime { get; set; }
         [Required]
         [StringLength(255)]
        public string Enroll { get; set; }
        [Required]
         [StringLength(255)]
        public string Type { get; set; }
       [Required]
        public int ClientId { get; set; }
        public Client Client { get; set; }
        [Required]
        public DateTime MonitoringDate { get; set; }
        [Required]
        public DateTime NextMonitoringDate { get; set; }
        [Required]
        [StringLength(255)]
        public string Monitoringtype { get; set; }
        public int? PlacementId { get; set; }
    }
}

File: spring-backend/src/main/java/com/mtp/api/models/Monitoring.java
package com.mtp.api.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "Monitorings")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Monitoring {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    @Size(max = 50)
    private String monitoringTime;

    @NotNull
    @Size(max = 255)
    private String enroll;

    @NotNull
    @Size(max = 255)
    private String type;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ClientId")
    private Client client;

    @NotNull
    private LocalDateTime monitoringDate;

    @NotNull
    private LocalDateTime nextMonitoringDate;

    @NotNull
    @Size(max = 255)
    private String monitoringtype;

    private Integer placementId;
}
